homeassistant/components/conversation/default_agent.py

```diff
@@ -1144,18 +1111,196 @@ async def _match_triggers(self, sentence: str) -> SentenceTriggerResult | None:
 
         return SentenceTriggerResult(sentence, matched_template, matched_triggers)
 
+    async def _handle_trigger_result(
+        self, result: SentenceTriggerResult, user_input: ConversationInput
+    ) -> str:
+        """Run sentence trigger callbacks and return response text."""
+
+        # Gather callback responses in parallel
+        trigger_callbacks = [
+            self._trigger_sentences[trigger_id].callback(
+                user_input.text, trigger_result, user_input.device_id
+            )
+            for trigger_id, trigger_result in result.matched_triggers.items()
+        ]
+
+        # Use first non-empty result as response.
+        #
+        # There may be multiple copies of a trigger running when editing in
+        # the UI, so it's critical that we filter out empty responses here.
+        response_text = ""
+        response_set_by_trigger = False
+        for trigger_future in asyncio.as_completed(trigger_callbacks):
```

Comment:
Since we know that none of these triggers do I/O, we would benefit here from just using `for… await` as those are run inline, without yielding. But that can be for another PR too.

Reply:
using `asyncio` helpers will guarantee a yield